lib/src/value/color.ts

```diff
@@ -30,102 +37,77 @@ export class SassColor extends Value {
   private lightnessInternal?: number;
   private readonly alphaInternal: number;
 
-  private constructor(color: RgbColor | HslColor) {
+  constructor(color: RgbColor | HslColor | HwbColor) {
     super();
 
     if ('red' in color) {
-      this.redInternal = color.red;
-      this.greenInternal = color.green;
-      this.blueInternal = color.blue;
+      this.redInternal = fuzzyAssertInRange(
+        Math.round(color.red),
+        0,
+        255,
+        'red'
+      );
+      this.greenInternal = fuzzyAssertInRange(
+        Math.round(color.green),
+        0,
+        255,
+        'green'
+      );
+      this.blueInternal = fuzzyAssertInRange(
+        Math.round(color.blue),
+        0,
+        255,
+        'blue'
+      );
+    } else if ('saturation' in color) {
+      this.hueInternal = color.hue % 360;
+      this.saturationInternal = fuzzyAssertInRange(
+        color.saturation,
+        0,
+        100,
+        'saturation'
+      );
+      this.lightnessInternal = fuzzyAssertInRange(
+        color.lightness,
+        0,
+        100,
+        'lightness'
+      );
     } else {
-      this.hueInternal = color.hue;
-      this.saturationInternal = color.saturation;
-      this.lightnessInternal = color.lightness;
+      // From https://www.w3.org/TR/css-color-4/#hwb-to-rgb
+      const scaledHue = (color.hue % 360) / 360;
```

Comment:
be careful. In JS, `-3 % 360` gives `-3`, not `357` (the sign of the return value matches the sign of the left operand, not of the right one, unlike in dart)

Reply:
Oh yikes, thanks for the heads up!